src/relay/transforms/fake_quantization_to_integer.cc

```diff
@@ -154,14 +154,30 @@ class SubgraphMutator : public ExprMutator {
     out_type_ = affine_types_[expr];
     static auto fqfq =
         Op::GetAttrMap<FTVMFakeQuantizationToInteger>("FTVMFakeQuantizationToInteger");
-    for (auto node : subgraph_) {
-      if (!fqfq.count(Downcast<Op>(node.as<CallNode>()->op))) {
-        // Only modify the subgraph if we have translation
-        // rules for every op
+    if (hard_fail_) {
+      for (auto node : subgraph_) {
+        const Op op = Downcast<Op>(node.as<CallNode>()->op);
+        ICHECK(fqfq.count(op) > 0)
+            << "Found no rewrite rule for " << AsText(op, false) << std::endl;
+      }
+      return Mutate(expr);
+    } else {
+      for (auto node : subgraph_) {
+        const Op op = Downcast<Op>(node.as<CallNode>()->op);
+        if (!fqfq.count(Downcast<Op>(op))) {
+          // Only modify the subgraph if we have translation
+          // rules for every op
+          DLOG(INFO) << "Found no rewrite rule for " << AsText(op, false) << std::endl;
```

Comment:
Can you move the if else for `hard_fail_` in here and use `LOG(FATAL)`, `ICHECK` just hits LOG_FATAL